rocAL/rocAL/include/meta_data/meta_data.h

```diff
@@ -234,28 +367,140 @@ struct BoundingBoxBatch: public MetaDataBatch
+                               getMetaDataBatchValues<std::vector<std::vector<int>>>(other, &MetaDataBatch::get_mask_polygons_count_batch).end());
+        _vertices_counts.insert(_vertices_counts.end(),
+                                getMetaDataBatchValues<std::vector<std::vector<std::vector<int>>>>(other, &MetaDataBatch::get_mask_vertices_count_batch).begin(),
+                                getMetaDataBatchValues<std::vector<std::vector<std::vector<int>>>>(other, &MetaDataBatch::get_mask_vertices_count_batch).end());
+        return *this;
+    }
+    void resize(int batch_size) override
+    {
+        _bb_cords.resize(batch_size);
+        _label_ids.resize(batch_size);
+        _info_batch.resize(batch_size);
+        _mask_cords.resize(batch_size);
+        _polygon_counts.resize(batch_size);
+        _vertices_counts.resize(batch_size);
+    }
+    void get_mask_cords_batch(std::vector<MaskCords>** mask_cords) override { *mask_cords = &_mask_cords; }
+    void get_mask_polygons_count_batch(std::vector<std::vector<int>>** polygon_counts) override { *polygon_counts = &_polygon_counts; }
+    void get_mask_vertices_count_batch(std::vector<std::vector<std::vector<int>>>** vertices_count) override { *vertices_count = &_vertices_counts; }
+    std::shared_ptr<MetaDataBatch> clone() override
+    {
+        return std::make_shared<InstanceSegmentationBatch>(*this);
+    }
+    void copy_data(std::vector<void*> buffer) override
+    {
+        if(buffer.size() < 2)
+            THROW("The buffers are insufficient") // TODO -change
+        int *labels_buffer = (int *)buffer[0];
+        double *bbox_buffer = (double *)buffer[1];
+        float *mask_buffer = (float *)buffer[2];
+        for(unsigned i = 0; i < _label_ids.size(); i++)
+        {
+            mempcpy(labels_buffer, _label_ids[i].data(), _label_ids[i].size() * sizeof(int));
+            memcpy(bbox_buffer, _bb_cords[i].data(), _label_ids[i].size() * sizeof(BoundingBoxCord));
+            memcpy(mask_buffer, _mask_cords[i].data(), _mask_cords[i].size() * sizeof(float));
+            labels_buffer += _label_ids[i].size();
+            bbox_buffer += (_label_ids[i].size() * 4);
+            mask_buffer += _mask_cords[i].size();
+        }
+    }
+    std::vector<size_t>& get_buffer_size() override
+    {
+        size_t size = 0;
+        for (auto label : _label_ids)
+            size += label.size();
+        _buffer_size.emplace_back(size * sizeof(int));
+        _buffer_size.emplace_back(size * 4 * sizeof(double));
+        size = 0;
+        for (auto mask : _mask_cords)
+            size += mask.size();
+        _buffer_size.emplace_back(size * sizeof(float));
+        return _buffer_size;
+    }
+protected:
+    std::vector<MaskCords> _mask_cords = {};
+    std::vector<std::vector<int>> _polygon_counts = {};
+    std::vector<std::vector<std::vector<int>>> _vertices_counts = {};
 };
 
-struct KeyPointBatch : public MetaDataBatch
+struct KeyPointBatch : public BoundingBoxBatch
```

Comment:
Please check the styling here. The way parathesis is used. It seems different at diff instances in the file